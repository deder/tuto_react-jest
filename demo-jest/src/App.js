import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import CardLarge from './card-large';
import CardSmall from './card-small';
const arrCard = [1,2,3,4,5,6,7,8,9,10]
class App extends Component {
  render() {
    return (
      <div className="App">
        <CardLarge/>
        { arrCard.map((card)=>(<CardSmall key={card}/>)) }
      </div>
    );
  }
}

export default App;
