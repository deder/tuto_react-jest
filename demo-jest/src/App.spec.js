import App from './App';

describe('<App />', ()=>{
    it("Snapshot", () => {
        const wrapper = shallow(
            <App />
        );
        expect(wrapper).to.matchSnapshot();
    });
    it('Contient 10 Composant CardSmall', () => {
        const wrapper = shallow(
            <App />
        );
        expect( wrapper.find('CardSmall') ).to.have.length(10); 
    });
    it('Contient 1 Composant CardLarge', () => {
        const wrapper = shallow(
            <App />
        );
        expect( wrapper.find('CardLarge') ).to.have.length(1); 
    });
})