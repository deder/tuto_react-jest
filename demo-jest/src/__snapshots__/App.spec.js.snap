// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Snapshot 1`] = `
<div
  className="App"
>
  <CardLarge />
  <CardSmall
    key="1"
  />
  <CardSmall
    key="2"
  />
  <CardSmall
    key="3"
  />
  <CardSmall
    key="4"
  />
  <CardSmall
    key="5"
  />
  <CardSmall
    key="6"
  />
  <CardSmall
    key="7"
  />
  <CardSmall
    key="8"
  />
  <CardSmall
    key="9"
  />
  <CardSmall
    key="10"
  />
</div>
`;
